src/Text/Pandoc/Readers/TikiWiki.hs

```diff
@@ -344,9 +344,13 @@ listItemLine nest = lineContent >>= parseContent
       return $ filterSpaces content ++ "\n" ++ Data.Maybe.fromMaybe "" continuation
     filterSpaces = reverse . dropWhile (== ' ') . reverse
     listContinuation = string (replicate nest '+') >> lineContent
+    removeSpace x = if B.toList x == [Space]
+                    then []
+                    else [x]
     parseContent x = do
       parsed <- parseFromString (many1 inline) x
-      return $ mconcat parsed
+      let ins = init parsed ++ removeSpace (last parsed)
```

Comment:
Generally I think we should avoid using partial functions like `init` and `last` whenever possible.  These crash with a runtime error when the list is empty.  Even if you know the list won't be empty when you write this code, sometimes a later change will cause the assumption to be violated, so it's good to code defensively.